File: 프로그래머스/네트워크.java
class Solution {
    static int answer = 0;

    public int solution(int n, int[][] computers) {
        boolean[] visited = new boolean[n];
        for(int i=0;i<computers.length;i++){
            for(int j=0;j<computers[i].length;j++){
                if(computers[i][j]==1 && !visited[j]){
                    answer++;
                    dfs(computers,j,visited);
                }
            }
        }
            
        return answer;
    }
    static void dfs(int[][] computers,int current,boolean[] visited){
        visited[current] = true;
        for(int i=0;i<computers.length;i++){
            if(computers[current][i] == 1 && !visited[i]){
                dfs(computers,i,visited);
            }
        }        

    }
}

File: 프로그래머스/네트워크.js
function solution(n, computers) {
    var answer = 0;
    const visited = new Array(n).fill(false);
    for(let i=0;i<computers.length;i++){
        if(!visited[i]){
            dfs(visited,i,computers,n);
            answer++;
        }
    }
    
    return answer;
}

const dfs = (visited,com,computers,n) => {
    visited[com] = true;
    for(let i=0;i<n;i++){
        if(computers[com][i] === 1 && !visited[i]){
            dfs(visited,i,computers,n);
        }
    }
}
